frontend/src/Frontend/UI/Dialogs/DeployConfirmation.hs

```diff
@@ -281,28 +289,36 @@ submitTransactionWithFeedback cmd chain nodeInfos = do
         -- Wait for the result
         doReqFailover clientEnvs (Api.listen Api.apiV1Client $ Api.ListenerRequest requestKey) >>= \case
           Nothing -> listen Status_Failed
-          Just (Api.ListenTimeout _i) -> listen Status_Failed
-          Just (Api.ListenResponse commandResult) -> case (Pact._crTxId commandResult, Pact._crResult commandResult) of
-            -- We should always have a txId when we have a result
-            (Just _txId, Pact.PactResult (Right a)) -> do
-              listen Status_Done
-              setMessage $ Just $ Right a
-              -- TODO wait for confirmation...
-            (_, Pact.PactResult (Left err)) -> do
+          Just res -> case res of
+            Left errs -> do
               listen Status_Failed
-              setMessage $ Just $ Left err
-            -- This case shouldn't happen
-            _ -> listen Status_Failed
+              for_ (nonEmpty errs) $ setMessage . Just . Left . packHttpErr . NEL.last
+            Right (Api.ListenTimeout _i) -> listen Status_Failed
+            Right (Api.ListenResponse commandResult) -> case (Pact._crTxId commandResult, Pact._crResult commandResult) of
+              -- We should always have a txId when we have a result
+              (Just _txId, Pact.PactResult (Right a)) -> do
+                listen Status_Done
+                setMessage $ Just $ Right a
+                -- TODO wait for confirmation...
+              (_, Pact.PactResult (Left err)) -> do
+                listen Status_Failed
+                setMessage $ Just $ Left $ NetworkError_CommandFailure err
+              -- This case shouldn't happen
+              _ -> listen Status_Failed
```

Comment:
Might be worth having something here to report that we've encountered an "impossible" case ?